jib-cli/src/main/java/com/google/cloud/tools/jib/cli/jar/JarProcessor.java

```diff
@@ -145,6 +149,33 @@ public static JarType determineJarType(Path jarPath) throws IOException {
     return layers;
   }
 
+  /**
+   * Computes the entrypoint for a standard jar in exploded mode.
+   *
+   * @param jarPath path to jar file
+   * @param tempDirPath path to temporary jib local directory
+   * @param layers list of {@link FileEntriesLayer}
+   * @return list of {@link String} representing entrypoint
+   * @throws IOException if I/O error occurs when opening the jar file or if temporary directory
+   *     provided doesn't exist
+   */
+  protected static ImmutableList<String> computeEntrypointForExplodedStandard(
+      Path jarPath, Path tempDirPath, List<FileEntriesLayer> layers) throws IOException {
+    Path localExplodedJarRoot = tempDirPath;
+    ZipUtil.unzip(jarPath, localExplodedJarRoot);
+
+    String mainClass = null;
+    try (JarFile jarFile = new JarFile(jarPath.toFile())) {
+      mainClass = jarFile.getManifest().getMainAttributes().getValue(Attributes.Name.MAIN_CLASS);
+    }
+    if (mainClass == null) {
+      throw new IllegalArgumentException(
+          "`Main-Class:` attribute to define an application main class not defined in the input Jar's manifest (`META-INF/MANIFEST.MF` in the JAR).");
```

Comment:
Sorry, I think this reads better:
```suggestion
          "`Main-Class:` attribute for an application main class not defined in the input Jar's manifest (`META-INF/MANIFEST.MF` in the Jar).");
```